torch/distributed/distributed_c10d.py

```diff
@@ -1327,6 +1327,21 @@ def _get_split_source(pg):
 
     return split_from
 
+def _shutdown_backend(pg):
+    """
+    Try to shut down the backend of a process group.
+    Currently, only ProcessGroupNCCL backend is supported.
```

Comment:
wondering if it makes sense to plumb the shutdown call into all backends and let them be no-ops in the c++ layer, but i guess this is fine for now, and my idea is just more work